cypress/e2e/patient_spec/patient_department.cy.ts

```diff
@@ -0,0 +1,101 @@
+    cy.loginByApi("devdepartment");
+    cy.visit("/");
+    facilityCreation.selectFacility("GHC Payyanur");
+  });
+
+  const departmentName = generateDeptName();
+
+  it("Create a new department/team and a sub-department/team, then verify that the search functionality works correctly on the dashboard", () => {
+    const subDepartmentName = generateDeptName("Sub-");
+    const description = generateRandomCharacter({
+      charLimit: 50,
+    });
+    const OrganizationType = "Department";
+    const updatedOrganizationType = "Team";
+    const updatedDescription = generateRandomCharacter({
+      charLimit: 50,
+    });
+    const updatedDepartmentTeamName = generateDeptName("Updated-Sub-");
+    patientDepartments
+      .navigateToSettings()
+      .navigateToDepartments()
+      .clickAddDepartmentTeam()
+      .enterName(departmentName)
+      .selectType(OrganizationType)
+      .enterDescription(description)
+      .interceptCreateRequest()
+      .clickCreateOrganization()
+      .verifyCreateRequest()
+      .assertCreationSuccess()
+      .searchDepartmentTeam(departmentName)
+      .openDepartmentsTeamDetails()
+      .clickAddDepartmentTeam()
+      .enterName(subDepartmentName)
+      .selectType(OrganizationType)
+      .enterDescription(description)
+      .interceptCreateRequest()
+      .clickCreateOrganization()
+      .verifyCreateRequest()
+      .assertCreationSuccess()
+      .searchDepartmentTeam(subDepartmentName)
+      .verifyDepartmentTeamContentInList(subDepartmentName, OrganizationType)
+      .clickEditOrganization()
+      .enterName(updatedDepartmentTeamName)
+      .selectType(updatedOrganizationType)
+      .enterDescription(updatedDescription)
+      .interceptUpdateRequest()
+      .clickUpdateOrganization()
+      .verifyUpdateRequest()
+      .assertUpdateSuccess()
+      .searchDepartmentTeam(updatedDepartmentTeamName)
+      .verifyDepartmentTeamContentInList(
+        updatedDepartmentTeamName,
+        updatedOrganizationType,
+      );
+  });
+
+  it("Navigate to the facility's administration department and link a user to the facility", () => {
+    patientDepartments
+      .navigateToSettings()
+      .navigateToDepartments()
+      .searchDepartmentTeam("Administration")
+      .openDepartmentsTeamDetails()
+      .clickUsersTab()
+      .clickLinkUser()
+      .selectAssignedUser("devnurse3")
+      .selectRoleOfUser("Nurse")
+      .clickAddUserToOrganization()
+      .assertUserAddedSuccess()
+      .searchUser("devnurse3")
+      .clickEditRole()
+      .clickRemoveUser()
+      .clickConfirmRemove()
+      .assertUserRemovalSuccess();
+  });
+
+  it("Assign Department/Team to an Encounter and verify it", () => {
+    patientEncounter
+      .navigateToEncounters()
+      .clickInProgressEncounterFilter()
+      .openFirstEncounterDetails();
+    patientDepartments
+      .clickAddOrganization()
+      .selectAllOrganizationsTab()
+      .selectOrganization(departmentName)
```

Comment:
```suggestion
      .selectOrganization(departmentName)
```

- "Use a department name that already exists in the facility; do not use a randomly generated one from the tests above, as a failure in those tests could cause this test to fail as well. Each test should be independent